### ISSUE TITLE
[Accessibility issue] Inline Switch labels create unreliable results for screen readers

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#73641]
**Steps to reproduce**:
Go to StackBlitz ([editor](https://stackblitz.com/edit/react-ts-wfe6en?file=App.tsx) view or [app view](https://react-ts-wfe6en.stackblitz.io/)). Turn on screen reader and navigate through the different switches. (Chrome and VoiceOver combination creates some of the worst results).

**Actual Result**:
The associated label with the switch is unpredictable with what is read out depending on your Screen Reader and/or browser. 

**Expected Result**
Behaviour should be the same and predictable throughout.
